Move and recheck progress

sledgehammer999 opened the pull request:
See individual commit messages for explanation.

~~**Attention**
The 1st commit was made on a false assumption. It turns out that libtorrent currently doesn't report the move progress, so I opened https://github.com/arvidn/libtorrent/issues/2923.
Please review as if the feature exists. I won't merge current PR before libtorrent adds the feature.
If the feature is denied I'll modify this PR too.~

sledgehammer999 commented:
FYI, I am not 100% sure about the WebUI part.

glassez commented on src/base/bittorrent/torrenthandle.cpp:
```diff
@@ -521,15 +521,20 @@ int TorrentHandle::piecesHave() const
 
 qreal TorrentHandle::progress() const
```
I think it's useful to have two getters, one for current operation progress, and the second for download progress. Some kind of remote clients haven't store state.

sledgehammer999 commented:
Pushed a revised version:
1. @glassez I implemented a different getter
2. I dropped the progress for relocating. Only new status. My request on libtorrent's tracker for move progress isn't answered yet.
3. I don't know how to make the WebUI use taskProgress instead of progress when checking. (or mirror what I do on the GUI)

Consider reviewing this for merging. I am not waiting for "move progress" being implemented by libtorrent.

Chocobo1 commented on src/webui/www/private/scripts/dynamicTable.js:
```diff
@@ -814,6 +814,7 @@ var TorrentsTable = new Class({
                 case "checkingUP":
                 case "queuedForChecking":
                 case "checkingResumeData":
+                    case "moving":
```
indentations are off in this file

sledgehammer999 replied:
Sorry. It was a rebase conflict which was auto-solved.
I'll fix it later.

sledgehammer999 commented on src/base/bittorrent/torrenthandle.cpp:
```diff
@@ -1442,8 +1455,9 @@ void TorrentHandle::handleStorageMovedAlert(libtorrent::storage_moved_alert *p)
 
-    qDebug("Torrent is successfully moved from %s to %s"
-        , qUtf8Printable(m_moveStorageInfo.oldPath), qUtf8Printable(m_moveStorageInfo.newPath));
+    LogMsg(QCoreApplication::translate(i18nContext, "Successfully moved torrent: %1. New path: %2")
```
Are you sure? Personally I don't think so. I added this log line to have an indication of the move success.

Chocobo1 replied:
>Are you sure? 

just checked, seems we already has another log for this, so disregard my comment.

sledgehammer999 commented:
Indentation issue fixed.

glassez commented on src/base/bittorrent/torrenthandle.cpp:
```diff
@@ -527,11 +527,19 @@ qreal TorrentHandle::progress() const
     return progress;
 }
 
+qreal TorrentHandle::taskProgress() const
```
Unfortunately, now I have noticed that the current `progress()` method does not follow the download progress when the "check" occurs, but actually returns the checking progress. So this additional method turns out to be useless.

sledgehammer999 replied:
>the current progress() method does not follow the download progress when the "check" occurs

In my experience, the progress() will return the same value as before the recheck until the recheck finishes and updates which pieces we have. I observed this for torrents that I manually do a recheck.
Broken torrents probably won't return something in progress() until the recheck has finished.

In any case, should I drop this new method and incorporate it again into progress()?

sledgehammer999 commented:
PR updated

glassez commented on src/base/bittorrent/torrenthandle.cpp:
```diff
@@ -816,7 +822,10 @@ void TorrentHandle::updateState()
     else {
-        if (m_session->isQueueingSystemEnabled() && isQueued() && !isChecking()) {
+        if (isMoveInProgress()) {
+            m_state = TorrentState::Moving;
```
I don't understand why paused torrent can't be in moving state... Can't we move it? I don't see such restrictions in qBittorrent code.

sledgehammer999 commented:
>I don't understand why paused torrent can't be in moving state... Can't we move it? I don't see such restrictions in qBittorrent code.

You're correct. I must have been confused with the has recheck.
Fixed now.

glassez commented on src/base/bittorrent/torrenthandle.cpp:
```diff
@@ -519,15 +519,20 @@ int TorrentHandle::piecesHave() const
+        Q_ASSERT((progress >= 0.f) && (progress <= 1.f));
+        return progress;
+    }
+    else {
```
Please don't add "else" branch if "if" branch ends with "return" statement.

sledgehammer999 replied:
Fixed. PR updated.

sledgehammer999 commented:
Thx
